### PR TITLE
Updating smell test

### PR DESCRIPTION
**Pull request by nagarev**
## Description

Update Sonar Configurations

## Motivation and Context

Sonar configurations should be updated in order to take advantage of SonarCloud, taking into consideration the recommended way of doing it for [gradle projects](https://docs.sonarsource.com/sonarcloud/advanced-setup/ci-based-analysis/sonarscanner-for-gradle/), adapting it to the context of Github Actions.

## How Has This Been Tested?

By adding a test function plagued by bad practices and with no test coverage (now removed) to confirm that the SonalCloud analysis worked as expected:

Screenshot 2024-11-10 at 11 46 31 AM

Screenshot 2024-11-10 at 11 26 28 AM

Screenshot 2024-11-10 at 11 27 01 AM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

N/A

### CODE REVIEW

File: rskj-core/src/main/java/co/rsk/rpc/modules/debug/DebugModuleImpl.java [hunk 1]
```diff
@@ -143,6 +143,16 @@ private JsonNode traceBlock(Block block, TraceOptions options) {
...
+    private boolean testFunctionNotCovered() {
+        boolean var1 = false;
+        boolean var2 = true;
+        boolean var3 = false;
+        if (!var1 && var2 || !var3) {
+            return true;
+        }
+        return false;
+    }
+
```
Comment (fmacleal) [inline rskj-core/src/main/java/co/rsk/rpc/modules/debug/DebugModuleImpl.java:155]: question:

What is the meaning for each of this var? Can you explain it? What do you think in rename it to a more explanatory one?

Comment (nagarev) [reply rskj-core/src/main/java/co/rsk/rpc/modules/debug/DebugModuleImpl.java:155]: Ah! this code will be removed, it was only to test if the cloud would report the missing tests. I'll remove this as soon as we decide if we're ok with the changes. 💪🏼

Comment (Vovchyk) [reply rskj-core/src/main/java/co/rsk/rpc/modules/debug/DebugModuleImpl.java:155]: @nagarev , it seems like it's working... do you think we could remove it now and get the final version that can be merged to master?

Comment (fmacleal) [reply rskj-core/src/main/java/co/rsk/rpc/modules/debug/DebugModuleImpl.java:155]: In that case, I am approving it, since this code will be removed.

**Review by fmacleal — APPROVED** [review]
LGTM, good job!

File: rskj-core/src/main/java/co/rsk/rpc/modules/debug/DebugModuleImpl.java (hunk 1 above)
Comment (nagarev) [reply rskj-core/src/main/java/co/rsk/rpc/modules/debug/DebugModuleImpl.java]: Test function was removed. 💪🏼

**Review by fmacleal — APPROVED** [review]
LGTM

**Review by fmacleal — APPROVED** [review]

**Review by Vovchyk — APPROVED** [review]

**Pull request merged**